File: 372 - Android/Ashman/app/src/main/java/com/benedetto/lars/ashman/SettingsFragment.java
package com.benedetto.lars.ashman;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.support.v4.app.Fragment;

/**
 * A simple {@link Fragment} subclass.
 */
public class SettingsFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {


    public SettingsFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.preferences);
        SharedPreferences prefs = getPreferenceScreen().getSharedPreferences();
        onSharedPreferenceChanged(prefs, "startGhosts");
        onSharedPreferenceChanged(prefs, "ghostsPerLevel");
    }

    @Override
    public void onResume() {
        super.onResume();
        getPreferenceScreen().getSharedPreferences().registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        Preference pref = findPreference(key);
        pref.setSummary(Integer.toString(sharedPreferences.getInt(key, 2)));
    }
}

File: 372 - Android/Ashman/app/src/main/java/com/benedetto/lars/ashman/SettingsFragment.kt
package com.benedetto.lars.ashman

import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceFragment
import android.support.v4.app.Fragment


/**
 * A simple [Fragment] subclass.
 */
class SettingsFragment : PreferenceFragment(), SharedPreferences.OnSharedPreferenceChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.preferences)
        val prefs = preferenceScreen.sharedPreferences
        onSharedPreferenceChanged(prefs, "startGhosts")
        onSharedPreferenceChanged(prefs, "ghostsPerLevel")
    }

    override fun onResume() {
        super.onResume()
        preferenceScreen.sharedPreferences.registerOnSharedPreferenceChangeListener(this)
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        val pref = findPreference(key)
        pref.summary = Integer.toString(sharedPreferences.getInt(key, 2))
    }
}// Required empty public constructor

